pkg/aws/aws_test.go

```diff
@@ -60,7 +66,7 @@ func TestGenerateEnvironment(t *testing.T) {
 
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			plugin := aws.New(tc.region, &mockSTSAPI{expected: tc.credentials}, tc.sessionDuration)
+			plugin := aws.New(tc.region, &mockSTSAPI{expected: tc.credentials}, tc.sessionDuration, "")
```

Comment:
I struggle parsing Go, but is this really enough testing for this feature?  I feel like there's at least one big conditional on `if roleARN != ""` (not to mention all of the `err` checks), which makes me feel like there should be at least another test showing that both branches behave correctly, no?

Reply:
Yeah, I need to do that better. Given the current test coverage state, I was going to put that together in a separate PR to get some broad coverage

Reply:
See #57